### DISCUSSION TITLE
Creating a taxonomy for the models in model-zoo

### DISCUSSION THREAD

**Discussion by vikashg** [Project-MONAI/model-zoo#440] [Ideas]
Hello everyone, 
I am proposing to create a taxonomy for the model zoo. The idea is that model contributors do not need to do everything (that is train, deploy, multi-gpu support, evaluation, unit-testing) Even when browsing the models, people can at once see which models in the model zoo has what components and this will help them to navigate the examples for creating their own models. I am attaching here a prototype page. We need to choose different and more relevant icons. 
![modelzoo](https://github.com/Project-MONAI/model-zoo/assets/3863212/4596072c-6c3f-45fe-ad02-141b7ab574ef)

Happy to take questions. @yiheng-wang-nv

**Comment by yiheng-wang-nv** [comment]
Thanks for posting it, Hi @zephyrie , could you please also help to look at it?

**Reply by zephyrie** [reply]
+ @Nic-Ma @wyli @daguangxu @ericspod 

Hi @vikashg,

To improve the search and discoverability of models in the MONAI Model Zoo on GitHub, I suggest implementing a tagging system. Models can be tagged with information like modality (Pathology, Radiology, CT, MRI), task (Segmentation, Classification), dataset (BraTS, MedNIST), and workflow (Train, Multi-GPU Train, Deploy, Unit Tests).

I would suggest adding the workflow tags directly to the tagging system instead of using icons because it would limit the need to create description icons for the flows and make sure that everyone understands them.  Also, if we need to add more in the future, it can easily be added with additional tags.

By using a single "tags" field in the metadata.json file for each model, users can easily filter and search based on their specific requirements. Although, we'll need to be conscious of how many tags are displayed on the model card, or if we only want the full list of tags available within the pop-up modal.

I've already started working on an example implementation incorporating tagging and search functionalities. You can see an example here:

(I used auto-generated tags as an example, these aren't the tags that would actually be used for those specific models)
![Screenshot 2023-06-08 190601](https://github.com/Project-MONAI/model-zoo/assets/2168510/4363629e-dda1-4c6f-afc1-64da6bf03f27)

Let me know your thoughts on this approach!

**Reply by vikashg** [reply]
Hi @zephyrie This sounds even better than what I suggested. 

Thanks 
Vikash

**Reply by ericspod** [reply]
I think this is the way to go. The metadata.json file should have in it all the ancillary (meta) information about the model such as these tags. The modality in the network IO is mentioned but we can propose further optional attributes people can provide. Nice one @zephyrie!
